tests/test_check_stac_metadata.py

```diff
@@ -589,37 +406,36 @@ def should_raise_exception_if_non_s3_url_is_passed() -> None:
 
 
 def should_collect_assets_from_validated_collection_metadata_files(subtests: SubTests) -> None:
+    # Given one asset in another bucket and one relative link
     base_url = any_s3_url()
     metadata_url = f"{base_url}/{any_safe_filename()}"
     stac_object = deepcopy(MINIMAL_VALID_STAC_COLLECTION_OBJECT)
-    first_asset_url = f"{base_url}/{any_safe_filename()}"
+    first_asset_url = f"{any_s3_url()}/{any_safe_filename()}"
```

Comment:
Should this test include assets in the same directory as the root metadata file.